Errors during plugin require() are masked

When a plugin throws an error when it's required, postcss-cli always prints `Plugin Error: Cannot find module '${plugin}'` rather than reporting the actual error.

I spent a little while today debugging because of this, as I was getting the above error due to cssnano throwing a BrowserslistError due to a bad config. That root cause would have been significantly easier to figure out if the original error had not been masked.

--- a/index.js
+++ b/index.js
@@ -34,7 +34,10 @@ const cliConfig = {
         try {
           return require(plugin)()
         } catch (e) {
-          return error(`Plugin Error: Cannot find module '${plugin}'`)
+          const msg = e.message || `Cannot find module '${plugin}'`
+          let prefix = msg.includes(plugin) ? '' : ` (${plugin})`
+          if (e.name && e.name !== 'Error') prefix += `: ${e.name}`
+          return error(`Plugin Error${prefix}: ${msg}'`)
         }
       })
     : []

--- a/test/error.js
+++ b/test/error.js
@@ -40,7 +40,7 @@ test.failing('invalid --config', t => {
   })
 })
 
-test('PluginError', t => {
+test('plugin not found', t => {
   return cli(['test/fixtures/a.css', '-u', 'postcss-plugin', '-o', tmp()]).then(
     ({ err, code }) => {
       t.is(code, 1, 'expected non-zero error code')
@@ -50,6 +50,19 @@ test('PluginError', t => {
       )
     }
   )
+})
+
+test('plugin throws on require', t => {
+  return cli([
+    'test/fixtures/a.css',
+    '-u',
+    './test/fixtures/bad-plugin',
+    '-o',
+    tmp()
+  ]).then(({ err, code }) => {
+    t.is(code, 1, 'expected non-zero error code')
+    t.regex(err.toString(), /Plugin Error \(.*bad-plugin\): This fails/)
+  })
 })
 
 test('CssSyntaxError', t => {

--- a/test/fixtures/bad-plugin.js
+++ b/test/fixtures/bad-plugin.js
@@ -1,0 +1,1 @@
+throw new Error('This fails')

